src/GroupCurveChart/groupCurveChart.service.ts

```diff
@@ -3,12 +3,17 @@ import { CurveGroup, Datum } from './groupCurveChart.types';
 export const getAreaData = (curveGroup: CurveGroup, scaleType: string) => {
   const area: number[][] = [];
 
-  curveGroup['lower1'].data.map((point, index) => {
+  curveGroup['lower1'].data.forEach((point, index) => {
     const areaPoint: number[] = [];
     areaPoint.push(point[0]);
     areaPoint.push(point[1]);
     areaPoint.push(curveGroup['upper1'].data[index][1]);
-    scaleType === 'log' ? !areaPoint.includes(0) && area.push(areaPoint) : area.push(areaPoint);
+    if (scaleType === 'log' && !areaPoint.includes(0)) {
+      area.push(areaPoint);
+    }
+    if (scaleType !== 'log') {
+      area.push(areaPoint);
```

Comment:
the `if` block is the same for both conditions. Should they be combined?

Reply:
@chrisdicaprio  good catch!